src/models/visualisation.py

```diff
@@ -1,17 +1,72 @@
 """
-This module contains visualisation functions for vector, raster and graph data.
+This module contains visualisation functions for GeoGraphs.
 """
-
+from __future__ import annotations
 from typing import Optional, List, Tuple, Callable
 
 import folium
 import geopandas as gpd
 import networkx as nx
 import shapely.geometry
 from src.constants import CHERNOBYL_COORDS_WGS84, UTM35N, CEZ_DATA_PATH
+from src.models import geograph
+
+
+class GeoGraphViewer:
+    """Class for viewing GeoGraph object"""
+
+    def __init__(self, map_type: str = "folium") -> None:
+        """Class for viewing GeoGraph object.
+
+        Args:
+            map_type (str, optional): defines which package is used for creating the
+                map. One of ["folium","ipyleaflet"]. Folium is more widely compatible
+                and may also run on google colab but the viewer functionality is
+                limited, whereas ipyleaflet won't run on google colab but offers the
+                full viewer functionality.
+                Defaults to "folium".
+        """
+        if map_type == "ipyleaflet":
+            raise NotImplementedError
+        self.map_type = map_type
+        self.widget = None
+
+    def _repr_html_(self) -> str:
+        """Return raw html of widget as string.
+
+        This method gets called by IPython.display.display().
+        """
+
+        if self.widget is not None:
+            return self.widget._repr_html_()  # pylint: disable=protected-access
+
+    def add_graph(self, graph: geograph.GeoGraph) -> None:
```

Comment:
Wrt the comment on polymorphism above:
Since you will have a coulpe of similar functionalities like `add_graph` , `_repr_html_` or `add_layer_control` I think polymorphism could really make sense and simply things here (e.g. you wouldn't need to do the `if self.map_type == XYZ` checks all the time and the code would be more easily seperable in Folium and IpyWidgets.

If you wanna go this route and want to briefly brush up on polymorphism & inheritance in python, I recommend reading [this article](https://alan-turing-institute.github.io/rsd-engineeringcourse/ch05construction/08objects.html) first. I guess the software architecture pattern to take inspiration from in this case would be a Strategy-type pattern (see [here](https://www.tutorialspoint.com/design_pattern/strategy_pattern.htm) or [here](https://alan-turing-institute.github.io/rsd-engineeringcourse/ch05construction/09patterns.html) for more info).
I think that could might make things easier further down the line.

Reply:
Yep definitely agree. I was thinking that it's starting to look a bit unwieldy, so polymorphism will definitely be the way to go. Thanks for the resources, very useful 👍